tests/spread/general/license-valid/task.yaml

```diff
@@ -7,20 +7,20 @@ prepare: |
   mkdir test-snap
   cd test-snap
   snapcraft init
-  
-  # License is currently base triggered only
-  if [[ "$SPREAD_SYSTEM" =~ ubuntu-18.04 ]]; then
-      echo "base: core18" >> snap/snapcraft.yaml
-  elif [[ "$SPREAD_SYSTEM" =~ ubuntu-16.04 ]]; then
-      echo "base: core16" >> snap/snapcraft.yaml
-  else
-      FAIL "Test not supported for $SPREAD_SYSTEM"
-  fi 
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "../snaps/$SNAP/snap/snapcraft.yaml"
+
   echo "license: MIT" >> snap/snapcraft.yaml
 
 restore: |
   rm -rf test-snap
 
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
```

Comment:
the entire test-snap directory is wiped, this shouldn't be needed

Reply:
Well that was clearly a blind copy and paste fail.

Fixed!